presto-main/src/test/java/com/facebook/presto/operator/scalar/AbstractTestFunctions.java

```diff
@@ -126,15 +128,27 @@ protected void assertInvalidFunction(String projection)
         }
     }
 
-    protected void assertInvalidFunction(String projection, String message)
+    protected void assertInvalidFunction(String projection, StandardErrorCode errorCode, String messagePattern)
     {
         try {
             evaluateInvalid(projection);
-            fail("Expected to throw an INVALID_FUNCTION_ARGUMENT exception with message " + message);
+            fail("Expected to throw a PrestoException with message matching " + messagePattern);
+        }
+        catch (PrestoException e) {
+            assertEquals(e.getErrorCode(), errorCode.toErrorCode());
+            assertTrue(e.getMessage().equals(messagePattern) || e.getMessage().matches(messagePattern));
```

Comment:
It would be nice to rethrow some trace of `e` when check fails like
```
catch (PrestoException e) {
  try {
    ... the checcks ...
  } catch (Throwable failure) {
      failure.addSuppressed(e);
      throw e;
  }
```